sycl/test-e2e/OneapiDeviceSelector/empty_input_string.cpp

```diff
@@ -0,0 +1,9 @@
+
+// ONEAPI_DEVICE_SELECTOR, when called with an empty string, should be
+// treated in the same manner when ONEAPI_DEVICE_SELECTOR is not set.
```

Comment:
What does documentation say? Should we update it as well? I think raising an exception would be fine too.

+ @steffenlarsen

Reply:
I also wonder if this is the right behavior.  I think a strict reading of the BNF grammar in the [documentation](https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md#oneapi_device_selector) says that this should be an error.  I think the grammar requires at least one term in the expression string.

I think you could also make an argument that the current behavior is correct.  It does sort of make sense that setting `ONEAPI_DEVICE_SELECTOR` to an empty string would cause no devices to be selected.  (I.e. calling `device::get_devices` would return an empty vector.)

I think both of these choices (error or select no devices) would make more sense than treating this the same as having `ONEAPI_DEVICE_SELECTOR` unset.

Reply:
I now think it would be better to throw a meaningful exception instead of not selecting any device at all.

@RaviNarayanaswamy if not a major change, could we update OpenMP to also throw an error on ONEAPI_DEVICE_SELECTOR=""? I could do the same in SYCL runtime.

Reply:
Does DPC++ throw an error for other env variables if they are set to NULL?

Reply:
SYCL_DEVICE_ALLOWLIST does not. But, many other env variables do throw an exception when given an empty string. In the case of ONEAPI_DEVICE_SELECTOR, as Greg mentioned, an empty string goes against the acceptable input grammar, so throwing an error would be more suitable, unless we also update the documentation.

Reply:
The behavior of OpenMP and SYCL in this case should be identical to avoid many problems.
```
Currently OpenMP ignore ONEAPI_DEVICE_SELECTOR if it is set to null
```

And what about space?

Reply:
@zettai-reido  Both OpenMP and DPC++ throws a parsing error on ONEAPI_DEVICE_SELECTOR=" " (space)

```
Target LEVEL_ZERO RTL --> Loaded library 'libomptarget.sycl.wrap.so':
terminate called after throwing an instance of 'sycl::_V1::exception'
  what():  ONEAPI_DEVICE_SELECTOR parsing error. Backend is required but missing from " "
Aborted (core dumped)
```

Reply:
Yes OpenMP and DPC++ should have used the same code.  Unfortunately OpenMP could not use DPC++ implementation as it was integrated into sycl.   I will change  OpenMP to report error if the ONEAPI_DEVICE_SELECTOR is not valid instead of  silently ignoring it.  Can you open a JIRA for this.
Hopefully the parsing of ONEAPI_DEVICE_SELECTOR will be done in UR when we switch to using it.